solidity/test/Create2.test.js

```diff
@@ -0,0 +1,134 @@
+const { expect } = require('chai');
+const { ethers, chainweb, artifacts } = require('hardhat');
+const crypto = require('crypto');
+const { getSigners } = require('./utils/utils');
+const { hexlify } = require('ethers');
+const { switchChain, getChainIds } = chainweb;
+
+const CREATE2_DEPLOYER = '0x4e59b44847b379578588920cA78FbF26c0B4956C';
+const CREATE2_BYTECODE = "0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3";
+
+describe('Deploy SimpleToken via CREATE2 factory', async function () {
+  const salt = '0x' + crypto.randomBytes(32).toString('hex');
+  const contracts = [];
+
+  before(async function () {
+    console.log(`Using salt: ${salt}`);
+    await chainweb.runOverChains(async (chain) => {
+      const signers = await getSigners(chain);
+      const deployer = signers.deployer;
+
+      // Setup Contract Deployment Parameters and compute init code
+      console.log(`Compute Contract Init Code`);
+      const SimpleToken = await ethers.getContractFactory('SimpleToken', deployer);
+      const constructorArgs = [ethers.parseUnits('1000000'), deployer.address];
+      const encodedConstructorArgs = SimpleToken.interface.encodeDeploy(constructorArgs);
+      const initCode = ethers.concat([SimpleToken.bytecode, encodedConstructorArgs]);
+
+      // Expected address
+      const address = ethers.getCreate2Address(
```

Comment:
This should be called `predictedAddress`. It gets set in the contract object, and it looks as if this is an actual deployed address. I had to keep going back to look at the before code to see what that address actually is.